SqlServer/Import-BacpacToSqlServer/Import-BacpacToSqlServer.psm1

```diff
@@ -103,17 +130,48 @@ function Import-BacpacToSqlServer
         {
             $connectionStringSegments = $ConnectionString.Split(';', [System.StringSplitOptions]::RemoveEmptyEntries)
 
-            $serverSegment = $connectionStringSegments | Where-Object { $PSItem.StartsWith('Data Source=') }
+            $serverSegment = $connectionStringSegments | Where-Object {
+                $PSItem.StartsWith('Data Source=') -or $PSItem.StartsWith('Server=')
+            }
             if (!([string]::IsNullOrEmpty($serverSegment)))
             {
                 $SqlServerName = $serverSegment.Split('=', [System.StringSplitOptions]::RemoveEmptyEntries)[1]
             }
 
-            $databaseSegment = $connectionStringSegments | Where-Object { $PSItem.StartsWith('Initial Catalog=') }
+            $databaseSegment = $connectionStringSegments | Where-Object {
+                $PSItem.StartsWith('Initial Catalog=') -or $PSItem.StartsWith('Database=')
+            }
             if (!([string]::IsNullOrEmpty($databaseSegment)))
             {
                 $DatabaseName = $databaseSegment.Split('=', [System.StringSplitOptions]::RemoveEmptyEntries)[1]
             }
+
+            $UsernameSegment = $connectionStringSegments | Where-Object { $PSItem.StartsWith('User Id=') }
+            if (!([string]::IsNullOrEmpty($UsernameSegment)))
+            {
+                $Username = $UsernameSegment.Split('=', [System.StringSplitOptions]::RemoveEmptyEntries)[1]
+            }
+
+            $PasswordSegment = $connectionStringSegments | Where-Object { $PSItem.StartsWith('Password=') }
+            if (!([string]::IsNullOrEmpty($PasswordSegment)))
+            {
+                $Password = $PasswordSegment.Split('=', [System.StringSplitOptions]::RemoveEmptyEntries)[1]
```

Comment:
Is it possible that the password may contain `=` character? I don't expect this for the other parameters, but it's plausible for a password. For example I went to a password generator site and `cA6i99=|Et"d` was among the results. You could use the overload with the `count` parameter:
![image](https://github.com/Lombiq/Utility-Scripts/assets/4007293/f2372313-6ce9-472e-95bb-44a1b070665f)

Reply:
Yes it is! And that overload works even if there are multiple `=` characters in the password, neat!